lib/kernel/src/code.erl

```diff
@@ -1867,6 +1867,11 @@ get_doc(Mod, #{sources:=[Source|Sources]}=Options) ->
                 ErtsDir ->
                     GetDoc(filename:join([ErtsDir, "ebin", atom_to_list(Mod) ++ ".beam"]))
             end;
+        cover_compiled ->
+            case cover_compiled_filename(Mod) of
+                {ok, Filename} -> GetDoc(Filename);
+                {error, missing} -> {error, missing}
+            end;
```

Comment:
It seems to me that there is no reason to re-implement searching for a BEAM file in the code path.

```suggestion
            case which(Mod, get_path()) of
                non_existing ->
                    {error, missing};
                Error when is_atom(Error) ->
                    {error, Error};
                Fn when is_list(Fn) ->
                    GetDoc(Fn)
            end;
```